### ISSUE TITLE
No training on my dataset 

### ISSUE THREAD

**Issue by irum** [YunYang1994/tensorflow-yolov3#281]
Hi YunYang,

I have tried to train model on my dataset but when I evaluate on test dataset . I am not getting any bounding boxes on the test dataset . Result.txt and predicted txt files are empty and after looking at TensorBoard I can see high confidence loss.

![71628824-C445-463B-801C-8F844B77E21A](https://user-images.githubusercontent.com/3971003/62895316-8815c800-bd46-11e9-9d01-f64b20a107aa.png)

Please can you guide me on this.
Thank you

**Comment by MichaelCong** [comment]
Hello, I want to ask a question. How to make the train.txt file during the training data set preparation process.thanks
E.g:
/home/yang/test/VOC/train/VOCdevkit/VOC2007/JPEGImages/000005.jpg 263,211,324,339,8 165,264,253,372,8 241,194,295,299,8
/home/yang/test/VOC/train/VOCdevkit/VOC2007/JPEGImages/000007.jpg 141,50,500,330,6
/home/yang/test/VOC/train/VOCdevkit/VOC2007/JPEGImages/000009.jpg 69,172,270,330,12 150,141,229,284,14 285,201,327,331,14 258,198,297,329,14

**Comment by fnando1995** [comment]
@MichaelCong Dont know if you still need this.
I just check this repo, really good.

For that annotation file you must built it by yourself. 

It is pretty easy to do. First if you check the  ./scripts/voc_annotations.py file you will understand that this file actually builds 2 files train.txt and test.txt. This Files will be used as direct connector to the images path and the boxes of each image. 

Each line of each of these files are IMAGE_PATH BOX,CLASS BOX,CLASS, .... (BOX is 4 numbers as x1,y1,x2,y2 or x,y,w,h havent check) either way is a way to illustrate the position of the box in the image.

The job of these files are to mix the path of the images to train or test with their own boxes. Easy cake.

With these 2 files, the training will initiate (if your are using own classes, dont forget to change classes.names file for the ones you will use).

regards,

**Comment by fnando1995** [comment]
@irum I will be trainning this implementation for a different class with own data. Hope to reproduce this and help you. will keep you up.

**Comment by fnando1995** [comment]
@irum I just finished the trainning. No problem happened.

**Comment by irum** [comment]
@fnando1995 
Thank you for the reply. Can you send the first 2 lines of your train.txt file
i.e path to .jpg min_x,miny,max_x, max_y .......

**Comment by fnando1995** [comment]
```
/home/path_to_a_folder_wherever_in_your_PC/images/img1.jpg 375,326,471,355,0 360,512,455,544,0 230,144,319,169,0 393,142,484,169,0 545,146,637,171,0
/home/path_to_a_folder_wherever_in_your_PC/images/img2.jpg 246,65,336,90,0 580,66,674,90,0 380,638,481,678,0 203,643,299,677,0 21,645,127,681,0
```


As you can see, this train.txt file have in each line:
`PATH_TO_IMAGE_IN_YOU_PC_1.jpg(space)x1a,y1a,x2a,y2a,class(space)x1b,y1b,x2b,y2b,class\n
PATH_TO_IMAGE_IN_YOU_PC_2.jpg(space)x1a,y1a,x2a,y2a,class\n`

where x1,y1,x2,y2 represent the box as the x,y values of the upper left and lower right points of the box. remember that each line represent an entire image, and each image could have more thatn 1 object to be detected or box, and all the boxes of a image should be in the same line separated by a blank space.

You may be confused of how to generate this file. For this you frist need to use a labeling program. I recommed you [this](https://github.com/tzutalin/labelImg) this program will help you boxing each image and generate a file with the annotations of the boxes of each image. 
After that concentrate the images in a folder and the annotation in other. Then use a script which reads the annotations files one by one and generate a single line in train.txt or test.txt (depends on your way to separate train and test data).

Hopes it helps.

kind regards

**Comment by irum** [comment]
@fnando1995 
Thank you very much for your quick reply and your time. I have fixed my problem.

**Comment by fnando1995** [comment]
Your welcome @irum 

If you can check how to freeze the ckpt to a .pb let me know please. In my way to check on that later for continue this repo.

**Comment by irum** [comment]
@fnando1995 
sure , no problem . I will let you know after initiating my training.

**Comment by HtutLynn** [comment]
Hey guys! I am also training a dataset-a portion of coco dataset. But the giou loss don't really decrease. All other loss decrease all the way down to zero but giou loss is really hard to decrease. after a few epochs, the total loss becomes nan. did the same case happen to you guys?
<img width="1060" alt="Screen Shot 2019-08-20 at 2 45 49 PM" src="https://user-images.githubusercontent.com/42746452/63330111-41812880-c359-11e9-917d-776034e0f9b5.png">

**Comment by fnando1995** [comment]
@HtutLynn 
Sorry man, actually that didnt happened to me in my training with own data (not coco/voc). Giou loss (something i think is related to IOU, dont really know) goes almost to zero. check. And no nan in my loss

![image](https://user-images.githubusercontent.com/14986529/63354744-a6715a00-c32a-11e9-86f0-28465296500e.png)

**Comment by HtutLynn** [comment]
@fnando1995  Okay, mate. By the way, what batch size did you use? Because giou loss in my case seems to be quite larger than yours. Also for training and making dataset class, did you use the exact code from this repo?

**Comment by HtutLynn** [comment]
@fnando1995 And as for the anchors, how did you produce the default anchor boxes for your own dataset? I think I have problem in that part.

**Comment by fnando1995** [comment]
@HtutLynn 
Hello!, sorry for the late answer.

1) I use batch size 8 with a gtx1080Ti. At 16 it collapse at certain point of the trainning.
2) I have used this and other repos for trainning. For creating the dataset i mentioned [THIS_REPO](https://github.com/tzutalin/labelImg) which I think is good; there are others great too, but i am good with this one because it uses XML which I consider is a great way to keep data organized (tags).
3) Actually, i havent change that, I think it is better for trainning, but not completly a must do. In fact, i have done little little search on that, so I can´t figure out what at the anchors exactly and how it outperfom the trainning if use custom data.

hope this will help. If you know better the anchor topic, please illustrate me :), mail/fb in the profile. 

Kind regards.
